Sources/Vapor/HTTP/Server/HTTPServer.swift

```diff
@@ -322,6 +404,60 @@ private final class HTTPServerConnection {
+    // If there is no socket path configured, don't do anything and report success. This will be called every-time on server shutdown.
+    guard let socketPath = configuration.unixDomainSocketPath else {
+        return eventLoop.makeSucceededFuture(())
+    }
+    
+    return threadPool.runIfActive(eventLoop: eventLoop) {
+        var socketFileStat = stat()
+        let (statResults, statError) = socketPath.withCString { path in
+            (stat(path, &socketFileStat), errno)
+        }
+        
+        // Since we only want to continue if there is a socket file, stop here if an error occurred with stat().
+        guard statResults == 0 else {
+            // If stat() failed because a file doesn't exist, then don't do anything and report success — the socket file won't need to be removed.
+            guard statError == ENOENT else {
+                // If stat reports another error, we likely can't remove the file either, which means NIO will fail when binding, so tell the user by logging it so they can take action.
+                configuration.logger.critical("Could not access an existing socket file located at \(socketPath): POSIX Error \(statError). Please remove this file in order to start the server.")
+                throw UnixDomainSocketPathError.inaccessible(IOError(errnoCode: statError, reason: "Could not access an existing socket file located at \(socketPath)"))
+            }
+            
+            // stat() reported that the file doesn't exist, indicating NIO can go ahead and create it when it binds, so don't do anything and report success.
+            return
+        }
+        
+        let mode = socketFileStat.st_mode
+        
+        // We only want to remove a socket file if it is actually a socket file - the user may have mistyped something, and it would be unfortunate for them to lose anything important otherwise.
+        guard mode&S_IFSOCK == S_IFSOCK else {
+            // A different type of file was found, so tell the user by logging it so they can take action.
+            configuration.logger.notice("An item at \(socketPath) already exists, but it is not a socket file. The socket file must not already exist before binding the server to a unix domain socket path.")
+            throw UnixDomainSocketPathError.unsupportedFile(mode, "An item at \(socketPath) already exists, but it is not a socket file")
+        }
+        
+        // The file exists, and is a socket file, so try to unlink it.
+        let (unlinkResults, unlinkError) = socketPath.withCString { path in
+            (unlink(path), errno)
+        }
+
+        guard unlinkResults == 0 else {
+            // If an error occurs during unlink, NIO will fail when binding, so report the error.
+            throw UnixDomainSocketPathError.couldNotRemove(IOError(errnoCode: unlinkError, reason: "Could not remove an existing socket file located at \(socketPath)"))
+        }
```

Comment:
Same `errno` concerns as earlier. Suggestion:
```suggestion
        guard unlink(socketPath) == 0 else {
            // If an error occurs during unlink, NIO will fail when binding, so report the error.
            throw UnixDomainSocketPathError.couldNotRemove(IOError(errnoCode: errno, reason: "Could not remove an existing socket file located at \(socketPath)"))
        }
```